src/app/regular-giving/regular-giving.component.ts

```diff
@@ -456,4 +524,32 @@ export class RegularGivingComponent implements OnInit, AfterViewInit {
 
     return !!this.paymentInfoErrorMessage;
   }
+
+  private validateGiftAidStep(): boolean  {
+    const errors: string[] = [];
+    if (typeof this.giftAid !== 'boolean') {
+      errors.push('Please choose whether you wish to claim Gift Aid.');
+    }
+
+    if (this.giftAid && !this.homeAddressFormValue) {
```

Comment:
For the long term I'm not too sure about doing these checks one by one. I think that Angular controls having validators attached – even if we have to add and remove them – might be less error prone. The built-in validators are also what we use elsewhere, so might better set us up for the possibility of reusable Form Groups, allowing different forms to share code for near-identical steps. Definitely not a blocker for now though.